src/lib/checkbox/checkbox.ts

```diff
@@ -291,6 +291,12 @@ export class MatCheckbox extends _MatCheckboxMixinBase implements ControlValueAc
 
     if (this._currentAnimationClass.length > 0) {
       element.classList.add(this._currentAnimationClass);
+
+      // Remove the animation class to avoid animation when the checkbox is moved between containers
+      const animationClass = this._currentAnimationClass;
+      setTimeout(() => {
```

Comment:
This one can run outside the `NgZone` since it's toggling the class directly on the element.

Reply:
Done. This changed the constructor.